lib/ReactViews/Errors/RaiseToUserErrorBoundary.tsx

```diff
@@ -0,0 +1,29 @@
+import i18next from "i18next";
+import React, { ErrorInfo } from "react";
+import ViewState from "../../ReactViewModels/ViewState";
+
+/**
+ * An error boundary that raises the error to the user.
+ */
+export default class RaiseToUserErrorBoundary extends React.Component<{
+  viewState: ViewState;
+}> {
+  state = { hasError: false };
+
+  static getDerivedStateFromError(error: any) {
+    return {
+      hasError: true
+    };
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    const newError = new Error(i18next.t("itemSearchTool.toolLoadError"));
```

Comment:
I think you should make the error message here configurable, otherwise it will always be ""Error loading search tool." even if this component isn't wrapping the search tool.

Reply:
Haha yes.. I have changed it to accept terria error options to customize the error message shown to user.